app/components/Sidebar/components/SharedDocumentLink.tsx

```diff
@@ -0,0 +1,139 @@
+import Collection from "~/models/Collection";
+import Document from "~/models/Document";
+import useStores from "~/hooks/useStores";
+import { NavigationNode } from "~/types";
+import Disclosure from "./Disclosure";
+import SidebarLink from "./SidebarLink";
+
+type Props = {
+  node: NavigationNode;
+  canUpdate: boolean;
+  collection?: Collection;
+  activeDocument: Document | null | undefined;
+  isDraft?: boolean;
+  depth: number;
+  index: number;
+  shareId: string;
+  parentId?: string;
+};
+
+function DocumentLink(
+  {
+    node,
+    canUpdate,
+    collection,
+    activeDocument,
+    isDraft,
+    depth,
+    shareId,
+    index,
+  }: Props,
+  ref: React.RefObject<HTMLAnchorElement>
+) {
+  const { documents } = useStores();
+  const { t } = useTranslation();
+
+  const isActiveDocument = activeDocument && activeDocument.id === node.id;
+
+  const hasChildDocuments =
+    !!node.children.length || activeDocument?.parentDocumentId === node.id;
+  const document = documents.get(node.id);
+
+  const showChildren = React.useMemo(() => {
+    return !!hasChildDocuments;
+  }, [hasChildDocuments]);
+
+  const [expanded, setExpanded] = React.useState(showChildren);
+
+  React.useEffect(() => {
+    if (showChildren) {
+      setExpanded(showChildren);
+    }
+  }, [showChildren]);
+
+  const handleDisclosureClick = React.useCallback(
+    (ev: React.SyntheticEvent) => {
+      ev.preventDefault();
+      ev.stopPropagation();
+      setExpanded(!expanded);
+    },
+    [expanded]
+  );
+
+  // since we don't have access to the collection sort here, we just put any drafts at the
+  // front of the list. this is slightly inconsisten with the logged-in behavior, but
+  // it's probably better to emphasize the draft nature of the document in a shared context
+  const nodeChildren = React.useMemo(() => {
+    if (
+      activeDocument?.isDraft &&
+      activeDocument?.isActive &&
+      activeDocument?.parentDocumentId === node.id
+    ) {
+      return [activeDocument?.asNavigationNode, ...node.children];
+    }
+
+    return node.children;
+  }, [
+    activeDocument?.isActive,
+    activeDocument?.isDraft,
+    activeDocument?.parentDocumentId,
+    activeDocument?.asNavigationNode,
+    node,
+  ]);
+
+  const title =
+    (activeDocument?.id === node.id ? activeDocument.title : node.title) ||
+    t("Untitled");
+
+  return (
+    <>
+      <SidebarLink
+        to={{
+          pathname: `/share/${shareId}${node.url}`,
+          state: {
+            title: node.title,
+          },
+        }}
+        label={
+          <>
+            {hasChildDocuments && (
+              <Disclosure expanded={expanded} onClick={handleDisclosureClick} />
+            )}
+            {title}
+          </>
+        }
+        depth={depth}
+        exact={false}
+        scrollIntoViewIfNeeded={!document?.isStarred}
+        isDraft={isDraft}
+        ref={ref}
+        isActive={() => {
+          return !!isActiveDocument;
+        }}
```

Comment:
this definition of `isActive` seems straightforward enough, but a little suspicious. what do you think @tommoor

Reply:
A little suspicious in that the `to` prop should match the location and mark as active automatically, you know the case in which that isn't happening?

Reply:
when you're at the root of a shared doc, there's no doc slug, so it doesn't match the `to` in the sidebar link. I started by overriding only that one case, but this code is simpler, otherwise it's 

```
return !!(isShareRoot ? isActiveDocument : match)
```